packages/block-library/src/cover/edit/index.js

```diff
@@ -103,7 +103,7 @@ function CoverEdit( {
 		'featured_media',
 		postId
 	);
-
+	const setCoverIsDark = useCoverIsDark( setAttributes );
```

Comment:
This could become `getCoverIsDark` and let the components update the attribute. What do you think?

Reply:
I was initially doing that and it just seemed like a lot of replicated:

```
__unstableMarkNextChangeAsNotPersistent();
setAttributes( { isDark: color.isDark } );
```
in each of the event handlers so moving to the hook removed the duplication, but I don't have a strong opinion on this. If you think this is better handled in the edit component I can move it back?

Reply:
If we can "batch" the attribute updates, I think it's okay to include this in the `undo` history and drop the "non-persistent" action call. But I see that most attribute updaters have wrappers like `setMedia`; that would mean unwrapping them as well, probably not worth it at the moment.

Reply:
batching the updates where possible was a much better approach!